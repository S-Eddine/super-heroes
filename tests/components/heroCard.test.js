import { shallow } from 'enzyme';
import React from 'react';
import HeroCard from '../../src/components/heroId/heroCard';

describe('heroCard', () => {
  it('renders correctly', () => {
    const heroId = { 
      comics : {
        items : ["comics"]
      },
      series : {
        items : ["series"]
      },
      thumbnail : {
        path: 'pathTest',
        extension: '.jpg'
      },
      name: "test",

    }
    const wrapper = shallow(<HeroCard heroId={heroId} />);
    expect(wrapper).toMatchSnapshot();
  });
});
