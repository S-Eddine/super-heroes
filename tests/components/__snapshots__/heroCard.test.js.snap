// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`heroCard renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(HeroCard)
    heroId={
      Object {
        "comics": Object {
          "items": Array [
            "comics",
          ],
        },
        "name": "test",
        "series": Object {
          "items": Array [
            "series",
          ],
        },
        "thumbnail": Object {
          "extension": ".jpg",
          "path": "pathTest",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "head": "HeroCard-head-4",
        "image": "HeroCard-image-2",
        "img": "HeroCard-img-3",
        "root": "HeroCard-root-1",
        "secondLevelHead": "HeroCard-secondLevelHead-5",
      },
      "heroId": Object {
        "comics": Object {
          "items": Array [
            "comics",
          ],
        },
        "name": "test",
        "series": Object {
          "items": Array [
            "series",
          ],
        },
        "thumbnail": Object {
          "extension": ".jpg",
          "path": "pathTest",
        },
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "head": "HeroCard-head-4",
          "image": "HeroCard-image-2",
          "img": "HeroCard-img-3",
          "root": "HeroCard-root-1",
          "secondLevelHead": "HeroCard-secondLevelHead-5",
        },
        "heroId": Object {
          "comics": Object {
            "items": Array [
              "comics",
            ],
          },
          "name": "test",
          "series": Object {
            "items": Array [
              "series",
            ],
          },
          "thumbnail": Object {
            "extension": ".jpg",
            "path": "pathTest",
          },
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
